### PR TITLE
feat: add Algolia integration for clone page

### PR DESCRIPTION
**Pull request by richard015ar**
Related issue: https://github.com/pressbooks/pressbooks/issues/2317

This PR integrates Instantsearch and Algolia libraries with a basic HTML structure to search, display and pick from the books list in our [Algolia's indexes used in Pressbooks Directory](https://github.com/pressbooks/pressbooks-book-directory-fe) and complete the `Book URL` field in the cloning page with a single click of a button.

To enable the feature, 3 ENV variables must be defined:
```
ALGOLIA_APP_ID=algolia_api_id
ALGOLIA_API_KEY=algolia_api_key
ALGOLIA_INDEX_NAME=algolia_index_name
```
You must get these values from the Algolia app.

Run `lando npm install` to install new dependencies.

### Test case
- Go to the clone page.
- You should see a new search input at the bottom.
- Type in the search box, and a list of books from the Directory should be listed at the bottom of the page in the form of book cards.
- Each book card contains a "Select" button. If you click there, the Book URL must be autocompleted with the selected Book URL.

### CODE REVIEW

File: assets/src/styles/clonerbookcard.scss [hunk 1]
```diff
@@ -0,0 +1,105 @@
+.ais-Hits-item {
```
Comment (richard015ar) [inline assets/src/styles/clonerbookcard.scss:1]: It was a compiled version of the current pressbooks network catalog style. I am sure better techniques and styles must be applied. Also, it does not meet the defined UI specs.

File: inc/admin/laf/namespace.php [hunk 2]
```diff
@@ -1161,6 +1161,28 @@ function ( $path ) use ( $assets ) {
 	wp_register_script( 'alpinejs', $assets->getPath( 'scripts/alpine.min.js' ), [], false, true );
 	wp_enqueue_script( 'alpinejs' );
 
+	if ( isset( $_REQUEST['page'] ) && $_REQUEST['page'] === 'pb_cloner' && \Pressbooks\Utility\are_algolia_search_env_var_presents() ) {
```
Comment (richard015ar) [inline inc/admin/laf/namespace.php:1164]: I don't think worth to create tests for this portion of the code. What's inside are all wp core methods already tested in the central core.

File: templates/admin/cloner-suggestions.html [hunk 3]
```diff
@@ -0,0 +1,34 @@
+<div class="book-card">
```
Comment (richard015ar) [inline templates/admin/cloner-suggestions.html:1]: More about this in https://www.algolia.com/doc/guides/building-search-ui/widgets/customize-an-existing-widget/js/#templating-your-ui

**Comment by SteelWagstaff** [comment]
@richard015ar how will open source users be able to use this feature? Do we have a way to hide the search feature or display a graceful fallback if the Algolia env variables are not entered?

**Pull request closed without merging**